nats-spring-cloud-stream-binder/src/main/java/io/nats/cloud/stream/binder/NatsMessageHandler.java

```diff
@@ -73,12 +74,9 @@ else if (payload instanceof String) {
 			return;
 		}
 
-		Object rt = message.getHeaders().get(NatsMessageProducer.REPLY_TO);
-		String replyTo = rt != null ? rt.toString() : null;
-		String subj = replyTo != null ? replyTo : this.subject;
-
-		if (this.connection != null && subj != null && subj.length() > 0) {
-			this.connection.publish(subj, bytes);
+		if (this.connection != null) {
+			final Object replyChannel = message.getHeaders().get(MessageHeaders.REPLY_CHANNEL);
+			this.connection.publish(this.subject, replyChannel != null ? replyChannel.toString() : null, bytes);
```

Comment:
Can you comment on what this change is, the 3 parameter publish instead of the 2

Reply:
I wrote that in the method's description:
> If the message contains a MessageHeaders.REPLY_CHANNEL header, it is passed on to allow taking advantage of build-in request/reply handling in Spring Integration / Spring Cloud Stream.

This puts Spring's `MessageHeaders.REPLY_CHANNEL` header as the `replyTo` name so that it can be used in `NatsMessageProducer`
```
headers.put(MessageHeaders.REPLY_CHANNEL, msg.getReplyTo());
```
which allows request/reply to work in Spring Integration/Spring Cloud Stream.

Reply:
As I'm looking at this code, would it be a correct assessment that this code is intended to circumvent / replace the built in request/reply mechanism in NATS and for Spring to do it's own thing?